src/MiniProfiler.Providers.PostgreSql/PostgreSqlStorage.cs

```diff
@@ -0,0 +1,489 @@
+        private readonly string MiniProfilerClientTimingsTableWithSchema;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PostgreSqlStorage"/> class with the specified connection string.
+        /// </summary>
+        /// <param name="connectionString">The connection string to use.</param>
+        /// <param name="schema">The schema that the tables are stored in.</param>
+        public PostgreSqlStorage(string connectionString, string schema = null) : base(connectionString)
+        {
+            Schema = schema;
+            if (Schema != null)
+            {
+                MiniProfilersTableWithSchema = $"{Schema}.{MiniProfilersTable}";
+                MiniProfilerTimingsTableWithSchema = $"{Schema}.{MiniProfilerTimingsTable}";
+                MiniProfilerClientTimingsTableWithSchema = $"{Schema}.{MiniProfilerClientTimingsTable}";
+            }
+            else
+            {
+                MiniProfilersTableWithSchema = MiniProfilersTable;
+                MiniProfilerTimingsTableWithSchema = MiniProfilerTimingsTable;
+                MiniProfilerClientTimingsTableWithSchema = MiniProfilerClientTimingsTable;
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PostgreSqlStorage"/> class with the specified connection string
+        /// and the given table names to use.
+        /// </summary>
+        /// <param name="connectionString">The connection string to use.</param>
+        /// <param name="profilersTable">The table name to use for MiniProfilers.</param>
+        /// <param name="timingsTable">The table name to use for MiniProfiler Timings.</param>
+        /// <param name="clientTimingsTable">The table name to use for MiniProfiler Client Timings.</param>
+        /// <param name="schema">The schema that the tables are stored in.</param>
+        public PostgreSqlStorage(string connectionString, string profilersTable, string timingsTable, string clientTimingsTable, string schema = null)
+            : base(connectionString, profilersTable, timingsTable, clientTimingsTable)
+        {
+            Schema = schema;
+            if (Schema != null)
+            {
+                MiniProfilersTableWithSchema = $"{Schema}.{MiniProfilersTable}";
+                MiniProfilerTimingsTableWithSchema = $"{Schema}.{MiniProfilerTimingsTable}";
+                MiniProfilerClientTimingsTableWithSchema = $"{Schema}.{MiniProfilerClientTimingsTable}";
+            }
+            else
+            {
+                MiniProfilersTableWithSchema = MiniProfilersTable;
+                MiniProfilerTimingsTableWithSchema = MiniProfilerTimingsTable;
+                MiniProfilerClientTimingsTableWithSchema = MiniProfilerClientTimingsTable;
+            }
+        }
+
+        private string _saveSql;
+        private string SaveSql => _saveSql ?? (_saveSql = $@"
+INSERT INTO {MiniProfilersTableWithSchema}
```

Comment:
On these - we'll just need to use the base fields, e.g. `MiniProfilersTableWithSchema` => `MiniProfilersTable`.